connectors/http_connector.go

```diff
@@ -4,40 +4,46 @@ import (
 	"bytes"
 	"encoding/json"
 	"fmt"
+	"io"
 	"net/http"
+	"net/url"
+
+	"github.com/nicksdlc/macaw/model"
 )
 
 // HTTPConnector is a connector to provide http infrastructure for the
 type HTTPConnector struct {
 	endpoint    string
+	port        uint16
 	httpClient  *http.Client
 	requestType string
 }
 
 // NewHTTPConnector creates new connector to send requests
-func NewHTTPConnector(endpoint, requestType string, client *http.Client) HTTPConnector {
-	return HTTPConnector{
+func NewHTTPConnector(endpoint, requestType string, port uint16, client *http.Client) *HTTPConnector {
```

Comment:
Namic looks a bit confusing. Connector should connect somewhere yet here we server is created, so it's rather Listener/Server.